builder/Dockerfile

```diff
@@ -33,31 +26,42 @@ RUN dnf -y update \
         libtool \
         libuuid-devel \
         make \
-        openssh-server \
         openssl-devel \
-        patchutils \
-        passwd \
         pkgconfig \
-        python2 \
-        rsync \
-        tar \
-        unzip \
         valgrind \
         wget \
-        which \
     && dnf clean all
 
 # Build dependencies from source
 WORKDIR ${BUILD_DIR}
 
-COPY builder builder
+COPY install builder/install
+COPY build builder/build
 COPY third_party third_party
 
-RUN "builder/install/install-dependencies.sh"
+RUN "builder/install/install-dependencies.sh" && \
+    rm -rf ${BUILD_DIR} && \
+    echo '/usr/local/lib' > /etc/ld.so.conf.d/usrlocallib.conf && ldconfig
 
-RUN rm -rf ${BUILD_DIR}
+FROM builder as dev
 
-RUN echo '/usr/local/lib' > /etc/ld.so.conf.d/usrlocallib.conf && ldconfig
+RUN dnf -y update \
+    && dnf -y install --nobest \
+        bison \
+        ca-certificates \
+        clang \
+        cracklib-dicts \
```

Comment:
```suggestion
        clang \
        clang-tools-extra \
        cracklib-dicts \
```
:wink:

Reply:
I was gonna add this alongside a few changes to improve development experience with `clangd`, but alright, we can add it.

Reply:
I am fine with this dependency being added later; I was just eager